src/CMakeLists.txt

```diff
@@ -345,6 +345,9 @@ add_subdirectory(runtime)
 # Build the Halide mono-header.
 ##
 
+set(CXX_COMPILER_IS_GNU_OR_CLANG $<CXX_COMPILER_ID:GNU,Clang,AppleClang>)
+set(CXX_COMPILER_IS_GNU_OR_CLANG_AND_NOT_CLANGCL $<AND:${CXX_COMPILER_IS_GNU_OR_CLANG},$<NOT:$<STREQUAL:${CMAKE_CXX_SIMULATE_ID},MSVC>>>)
+
```

Comment:
I'm very concerned about using `CMAKE_CXX_SIMULATE_ID`, which is undocumented, _especially_ without a comment. I wonder if the right place for ClangCL settings is in a toolchain or preset.

Reply:
Its actually documented since cmake 3.15, in the form of https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_SIMULATE_ID.html
I agree it could be promoted more and there is https://gitlab.kitware.com/cmake/cmake/-/issues/19724 that has the same complaint. But I doubt the variable will just go away now any time soon, since its used quite a bit with ClangCl and the Intel compiler. But I can add a comment if you like?

Reply:
Ahhhh `CMAKE_CXX_SIMULATE_ID` does not appear anywhere in the docs. I was mistaken about that.